packages/execution-environments/src/common/endowments/network.ts

```diff
@@ -2,6 +2,14 @@ import { allFunctions } from '../utils';
 
 type WebSocketCallback = (this: WebSocket, ev: any) => any;
 
+/**
+ * Create a network endowment, consisting of a `WebSocket` object and `fetch`
+ * function. This wraps the original implementations of `WebSocket` and `fetch`,
+ * to ensure that a bad actor cannot get access to the original objects.
```

Comment:
Also wraps the implementation to provide the teardown function such that we can cancel any pending requests, streams etc that may be open when a snap is terminated.

Reply:
Sorry if this was unclear, the main reason we do this is to provide the teardown function. May need to be reworded to reflect that. Because we want to make sure the teardown is effective we can't let bad actors touch the original objects 😄